resourcemanager/pool/workerpool/workerpool.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workerpool
+
+import (
+	"time"
+
+	"github.com/pingcap/tidb/resourcemanager"
+	"github.com/pingcap/tidb/resourcemanager/util"
+	tidbutil "github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/syncutil"
+	atomicutil "go.uber.org/atomic"
+)
+
+// Worker is worker interface.
+type Worker interface {
+	HandleTask(task any)
+	Close()
+}
+
+// WorkerPool is a pool of workers.
+type WorkerPool[T any] struct {
+	name           string
+	numWorkers     int32
+	runningWorkers atomicutil.Int32
```

Comment:
Running tasks?

Reply:
Yes， it means workers that running the task.